packages/expo/cli/start/server/metro/MetroTerminalReporter.ts

```diff
@@ -0,0 +1,223 @@
+
+const MAX_PROGRESS_BAR_CHAR_WIDTH = 16;
+const DARK_BLOCK_CHAR = '\u2593';
+const LIGHT_BLOCK_CHAR = '\u2591';
+/**
+ * Extends the default Metro logger and adds some additional features.
+ * Also removes the giant Metro logo from the output.
+ */
+export class MetroTerminalReporter extends TerminalReporter {
+  constructor(public projectRoot: string, terminal: Terminal) {
+    super(terminal);
+  }
+  /**
+   * Extends the bundle progress to include the current platform that we're bundling.
+   *
+   * @returns `iOS path/to/bundle.js ▓▓▓▓▓░░░░░░░░░░░ 36.6% (4790/7922)`
+   */
+  _getBundleStatusMessage(progress: BundleProgress, phase: BuildPhase): string {
+    const platform = getPlatformTagForBuildDetails(progress.bundleDetails);
+    const inProgress = phase === 'in_progress';
+
+    if (!inProgress) {
+      const status = phase === 'done' ? `Bundling complete ` : `Bundling failed `;
+      const color = phase === 'done' ? chalk.green : chalk.red;
+
+      const startTime = this._bundleTimers.get(progress.bundleDetails.buildID);
+      const time = chalk.dim(Date.now() - startTime + 'ms');
+      // iOS Bundling complete 150ms
+      return color(platform + status) + time;
+    }
+
+    const localPath = path.relative('.', progress.bundleDetails.entryFile);
+    const filledBar = Math.floor(progress.ratio * MAX_PROGRESS_BAR_CHAR_WIDTH);
+
+    const _progress = inProgress
+      ? chalk.green.bgGreen(DARK_BLOCK_CHAR.repeat(filledBar)) +
+        chalk.bgWhite.white(LIGHT_BLOCK_CHAR.repeat(MAX_PROGRESS_BAR_CHAR_WIDTH - filledBar)) +
+        chalk.bold(` ${(100 * progress.ratio).toFixed(1)}% `) +
+        chalk.dim(`(${progress.transformedFileCount}/${progress.totalFileCount})`)
+      : '';
+
+    return (
+      platform +
+      chalk.reset.dim(`${path.dirname(localPath)}/`) +
+      chalk.bold(path.basename(localPath)) +
+      ' ' +
+      _progress
+    );
+  }
+
+  _logInitializing(port: number, hasReducedPerformance: boolean): void {
+    // Don't print a giant logo...
+    this.terminal.log('Starting Metro Bundler');
+  }
+
+  shouldFilterClientLog(event: {
+    type: 'client_log';
+    level: 'trace' | 'info' | 'warn' | 'log' | 'group' | 'groupCollapsed' | 'groupEnd' | 'debug';
+    data: unknown[];
+  }): boolean {
+    return isAppRegistryStartupMessage(event.data);
+  }
+
+  /** Print the cache clear message. */
+  transformCacheReset(): void {
+    logWarning(
+      this.terminal,
+      chalk`Bundler cache is empty, rebuilding {dim (this may take a minute)}`
+    );
+  }
+
+  /** One of the first logs that will be printed */
+  dependencyGraphLoading(hasReducedPerformance: boolean): void {
+    // this.terminal.log('Dependency graph is loading...');
+    if (hasReducedPerformance) {
+      this.terminal.log(
+        chalk.red(
+          'Metro is operating with reduced performance.\n' +
+            'Please fix the problem above and restart Metro.\n\n'
```

Comment:
Add a comment that links/describes "the problem above" which is coupled to this code but in a different repo